daemon/network.go

```diff
@@ -315,9 +315,22 @@ func (daemon *Daemon) createNetwork(create types.NetworkCreateRequest, id string
 		driver = c.Config().DefaultDriver
 	}
 
+	networkOptions := make(map[string]string)
+	for k, v := range create.Options {
+		networkOptions[k] = v
+	}
+	if defaultOpts, ok := daemon.configStore.DefaultNetworkOpts[driver]; ok {
+		for k, v := range defaultOpts {
+			if _, ok := networkOptions[k]; !ok {
+				logrus.WithFields(logrus.Fields{"driver": driver, "network": id, k: v}).Debug("Applying network default option")
+				networkOptions[k] = v
+			}
+		}
+	}
```

Comment:
Defaults should only be applied to networks where `create.ConfigFrom == nil` as [`ConfigFrom` cannot be used in combination with driver options.](https://github.com/moby/moby/blob/0021339b9299c5f1596bdf772a42ce346f16f060/libnetwork/network.go#L382-L415)

Reply:
Hope I got that right! :+1: